skycoin/skywire
# Several improvements for the UI

Did you run `make format && make check`?
The go code was not changed. `npm run lint` and `npm run build` were used.

 Changes:	
- Now it is not possible to close the modal windows while the system is busy, to avoid incorrect expectations from the users about the state of the operations.

- Now the forms are deactivated while the system is busy.

- Now the rows of the visor list are links, which means that the option for opening the details in a news tab is now available.

- Now it is possible to close the confirmation dialogs by pressing esc.

- The old updater options were removed from the settings tab.

- Now the options for changing the password are not visible in the settings tab if the authentication features are not enabled in the back-end.

- Now the tooltips don't stay open if the mouse is over them, this prevents really annoying behaviors in small lists.

- Now the visor list has the click effect used in the list of the VPN UI. Also, the rows for offline visors are not highlighted while the mouse is over them.

How to test this PR:
- Use a modal window to create a transport and try to close it while the operation is in progrees. It should not be posible to close it and the form should be disabled during the operation.

- Select a row in the visor list, with the right mouse button. The menu should show the option for opening in a new tab.

- Use any option that needs confirmation. You should be able to close the confirmation modal window by pressing esc.

- Open the settings tab. It must not show the updater options. The password options should only appear if the auth options are active in the hypervisor.

- Open the server list in the VPN UI. Move the mouse from the top row to the bottom one, over the column with the options icons. You should see the tooltip saying "Options" for each icon and the tooltip should not interfere with the ability to display the tooltip of any other icon.
